src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.Helix.targets

```diff
@@ -0,0 +1,305 @@
+      Items="@(LocalPayloadFileForUpload->Distinct())"
+      Overwrite="$(OverwriteOnUpload)" />
+
+    <!-- Supplemental payload.  TODO: Can be combined with Correlation Payload items, there's not much(any?) difference -->
+    <UploadToAzure
+      ConnectionString="$(CloudDropConnectionString)"
+      ContainerName="$(ContainerName)"
+      Items="@(SupplementalPayload)"
+      Overwrite="$(OverwriteOnUpload)"
+      Condition="'@(SupplementalPayload)' != ''" />
+
+  </Target>
+
+  <Target Name="CreateTestListJson">
+    <!-- We always bring the supplemental payload folder.  
+         This contains whatever's in src\Microsoft.DotNet.Build.CloudTestTasks\RunnerScripts -->
+    <ItemGroup>
+      <CorrelationPayloadUri Include="@(SupplementalPayload->'$(DropUri)%(RelativeBlobPath)$(DropUriReadOnlyToken)')" />
+      <CorrelationPayloadUri Include="@(LocalPayloadFileForUpload->'$(DropUri)%(RelativeBlobPath)$(DropUriReadOnlyToken)')" />
+    </ItemGroup>
+    <PropertyGroup>
+      <!-- flatten it into a property as msbuild chokes on @(CorrelationPayloadUri) in FunctionalTest.CorrelationPayloadUris -->
+      <CorrelationPayloadUris>@(CorrelationPayloadUri)</CorrelationPayloadUris>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <HelixProcessedWorkItem>
+        <Command>%(Command)</Command>
+        <CorrelationPayloadUris>[$(CorrelationPayloadUris)]</CorrelationPayloadUris>
+        <PayloadUri>$(DropUri)%(RelativeBlobPath)$(DropUriReadOnlyToken)</PayloadUri>
+        <WorkItemId>%(WorkItemId)</WorkItemId>
+        <TimeoutInSeconds>%(TimeoutInSeconds)</TimeoutInSeconds>
+      </HelixProcessedWorkItem>
+    </ItemGroup>
+
+    <!-- I'd love a built-in way to do this, but I'm compromising here since this lets us use 
+         arbitrary extra metadata in the future to compose work items, then strip out extra stuff used along the way.
+         It lets us use a single item to bring along everything we might possibly need in the future -->
+    <RemoveItemMetadata Items="@(HelixProcessedWorkItem)" FieldsToRemove="RelativeBlobPath;PayloadFile">
+      <Output TaskParameter="ProcessedItems" ItemName="HelixProcessedWorkItemForList"/>
+    </RemoveItemMetadata>
+
+    <WriteItemsToJson JsonFileName="$(TestListFilename)" Items="@(HelixProcessedWorkItemForList)" />
+    <ItemGroup>
+      <HelixWorkItemList Include="$(TestListFilename)">
+        <RelativeBlobPath>$(TestListFilename)</RelativeBlobPath>
+        <BuildCompleteJson>./BuildComplete.json</BuildCompleteJson>
+        <OfficialBuildJson>./OfficialBuild.json</OfficialBuildJson>
+        <HelixJobUploadCompletePath>./helixjobuploadcomplete.sem</HelixJobUploadCompletePath>
+      </HelixWorkItemList>
+    </ItemGroup>
+
+    <UploadToAzure
+     ConnectionString="$(CloudDropConnectionString)"
+     ContainerName="$(ContainerName)"
+     Items="@(HelixWorkItemList)"
+     Overwrite="$(OverwriteOnUpload)" />
+  </Target>
+
+  <!-- Write event hub notification JSON files (one per queue)-->
```

Comment:
Correct, we want identical work item lists for each run since we're sending the same jobs there.  There are environment variables present on the machines should we ever want to differentiate behavior based on where we end up, but I oppose the idea.